ktor-utils/common/src/io/ktor/util/CaseInsensitiveMap.kt

```
@@ -5,75 +5,417 @@
+
+            override fun next(): String {
+                if (!hasNext()) throw NoSuchElementException()
+                val idx = insertionOrder[orderIndex]
+                lastKey = keyStorage[idx]!!
+                orderIndex++
+                advance()
+                return lastKey!!
+            }
+
+            override fun remove() {
+                val key = lastKey ?: throw IllegalStateException()
```

detekt (1.23.8)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)